### ISSUE TITLE
GAN example not working

### ISSUE THREAD

**Issue by vobecant** [PyTorchLightning/pytorch-lightning-bolts#291]
## 🐛 Bug

The GAN example for MNIST doesn't work as advised in the script basic_gan_module.py https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/gans/basic/basic_gan_module.py

### To Reproduce

To reproduce the issue just run https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/gans/basic/basic_gan_module.py
`basic_gan_module.py --gpus 1`

### Expected behavior

Train the GAN network well for MNIST dataset with the default setting.

### Environment
* CUDA:
	- GPU:
		- GeForce GTX 1080 Ti
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0
	- pytorch-lightning: 1.0.2
	- tqdm:              4.50.2
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.5
	- version:           #1 SMP Tue Aug 25 17:23:54 UTC 2020

**Comment by akihironitta** [comment]
@vobecant Thank you for submitting the issue! Would it be possible to share the error message with us?

**Comment by vobecant** [comment]
Traceback (most recent call last):
  File "basic_gan_module.py", line 203, in <module>
    dm, model, trainer = cli_main()
  File "basic_gan_module.py", line 172, in cli_main
    from pl_bolts.datamodules import CIFAR10DataModule, ImagenetDataModule, MNISTDataModule, STL10DataModule
ImportError: cannot import name 'ImagenetDataModule' from 'pl_bolts.datamodules' (/home/vobecant/miniconda3/envs/pytorch16/lib/python3.8/site-packages/pl_bolts/datamodules/__init__.py)

**Comment by akihironitta** [comment]
@vobecant Thank you for sharing the error message! I'll have a look at it.

**Comment by akihironitta** [comment]
@vobecant Currently, we have a problem in `pl_bolts/datamodules/__init__.py`, and the error occurs as you described. An easiest workaround would be to modify `cli_main()` function defined in `pl_bolts/models/gans/basic/basic_gan_module.py`.

Specifically, would you mind changing
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/f0939f62a5559a9a68cbb60ab8cbddac8c42cb05/pl_bolts/models/gans/basic/basic_gan_module.py#L171
to
```python
    from pl_bolts.datamodules import CIFAR10DataModule
    from pl_bolts.datamodules.imagenet_datamodule import ImagenetDataModule
    from pl_bolts.datamodules.mnist_datamodule import MNISTDataModule
    from pl_bolts.datamodules.stl10_datamodule import STL10DataModule
```
?
It should work, but let me know if it still doesn't work.

**Comment by akihironitta** [comment]
@vobecant It looks the problem in `pl_bolts/datamodules/__init__.py` has been solved in #303, so would you just update to the current master branch, and retry?

**Comment by ananyahjha93** [comment]
like @akihironitta mentioned, this was due to the datamodules issue which has been fixed now.
